hrjobcontract/js/src/job-contract/controllers/modal/modal-contract.js

```diff
@@ -53,6 +54,9 @@ define([
             angular.copy(entity,$scope.entity);
             angular.copy(files,$scope.files);
 
+            $scope.entity.details.period_start_date = convertToDateObject($scope.entity.details.period_start_date);
```

Comment:
Not sure I understand the comment correctly, but the `$filter` is used in the `convertToDateObject` function:
```js
function convertToDateObject(dateString) {
    var dateObj = $filter('formatDate')(dateString, Date);

    return dateObj !== 'Unspecified' ? dateObj : dateString;
}
```
the reason for this was because the filter still returns the very specific `Unspecified` string (instead of, say, `undefined`). I didn't want to change the return value of the filter itself and risking to introduce yet more breaking changes, so I'm using the `convertToDateObject` to make sure that the invalid (usually `null`) date is returned in case the filter returns `Unspecified`. The same logic is applied in some other components in other extensions